src/plugin.js

```diff
@@ -231,18 +260,33 @@ const onPlayerReady = (player) => {
  * to you; if not, remove the wait for "ready"!
  *
  * @function eme
+ * @param    {Object} [player]
+ *           A player object.
  * @param    {Object} [options={}]
  *           An object of options left to the plugin author to define.
  */
-const eme = function(options = {}) {
-  this.eme.options = options;
-
-  this.ready(() => onPlayerReady(this));
+const eme = function(player, options = {}) {
```

Comment:
wouldn't this be a breaking change for anyone using contrib-eme? We could do the same by doing:
`const player = this;`(and disabling the linting rule) inside the method

Reply:
I'm not following how this would be a breaking change? This function gets called in the plugin register https://github.com/videojs/videojs-contrib-eme/pull/61/files/fc14190f75edb07e4c6101211475d88b7d9c74c3#diff-a4f485aa834ca5b3c24819fceb39c56fR288 and is not the same as `player.eme()`

Reply:
true, I guess I don't see the point of adding the player argument when `this === player` in this function

Reply:
I believe the state of the eme plugin is carrying over after player dispose somehow, I saw this when updating the tests.

Reply:
hmm, I tried this out in isolation, and it seemed ok. Maybe something else caused the memory issue?

Reply:
uh, `this` instead of player to be clear